core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/ByteOrderedToken.java

```diff
@@ -29,10 +29,11 @@
 
   private final ByteBuffer value;
 
-  public ByteOrderedToken(ByteBuffer value) {
-    this.value = stripTrailingZeroBytes(value);
+  public ByteOrderedToken(@NonNull ByteBuffer value) {
+    this.value = ByteBuffer.wrap(Bytes.getArray(value)).asReadOnlyBuffer();
```

Comment:
Since the class is marked `@Immutable`, I'm creating a defensive, read-only copy of the input buffer.

Reply:
:+1: that's justified, the byte buffer can be provided directly by the end user (via `TokenMap.newToken`).